File: 2302-count-subarrays-with-score-less-than-k/2302-count-subarrays-with-score-less-than-k.java
class Solution {
    public long countSubarrays(int[] nums, long k) {
        long cnt=0; int j=0; long sum=0;
        for(int i=0;i<nums.length;i++)
        {
            sum+=nums[i];
            while(sum*(i-j+1)>=k)
            {
                sum-=nums[j];
                j++;
            }
             cnt+=i-j+1;
        }
        return cnt;
    }
}

File: 2302-count-subarrays-with-score-less-than-k/2302-count-subarrays-with-score-less-than-k.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& nums, long long k) {
        long long int sum=0; long long ans=0; long long j=0;
        for(int i=0;i<nums.size();i++)
        {
            sum+=nums[i]; 
            while(sum*(i-j+1)>=k)
            {
                sum-=nums[j];
                j++;
            }
            ans+=i-j+1;
        }
        return ans;
    }
};